applications/GeoMechanicsApplication/tests/cpp_tests/custom_processes/test_fix_water_pressures_above_phreatic_line.cpp

```diff
@@ -0,0 +1,245 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+
+#include "custom_processes/fix_water_pressures_above_phreatic_line.h"
+#include "geo_mechanics_application_variables.h"
+#include "tests/cpp_tests/geo_mechanics_fast_suite.h"
+
+namespace
+{
+using namespace Kratos;
+
+Parameters CreateParametersWithConstantPhreaticLineAtHeight(double height)
+{
+    return Parameters{R"(
+            {
+                "model_part_name": "foo",
+                "x_coordinates": [0.0],
+                "y_coordinates": [)" +
+                      std::to_string(height) + R"(],
+                "z_coordinates": [0.0]
+            }  )"};
+}
+
+ModelPart& CreateModelPartWithTwoNodesAtHeights(Model& rModel, double y_node_1, double y_node_2)
+{
+    auto& r_result = rModel.CreateModelPart("foo");
+    r_result.AddNodalSolutionStepVariable(WATER_PRESSURE);
+    r_result.AddNodalSolutionStepVariable(TOTAL_DISPLACEMENT);
+    r_result.CreateNewNode(1, 0.0, y_node_1, 0.0);
+    r_result.CreateNewNode(2, 0.0, y_node_2, 0.0);
+
+    for (auto& r_node : r_result.Nodes()) {
+        r_node.AddDof(WATER_PRESSURE);
+    }
+
+    return r_result;
+}
+
+} // namespace
+
+namespace Kratos::Testing
+{
+
+KRATOS_TEST_CASE_IN_SUITE(TestFixWaterPressureAbovePhreaticLine_ThrowsUponConstructionWhenXOrYCoordinatesAreEmpty,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arrange
+    constexpr double y_node_1     = 0.0;
+    constexpr double y_node_2     = -2.0;
+    auto             model        = Model{};
+    auto&            r_model_part = CreateModelPartWithTwoNodesAtHeights(model, y_node_1, y_node_2);
+
+    auto test_parameters = Parameters{R"(
+            {
+                "model_part_name": "foo",
+                "x_coordinates": [],
+                "y_coordinates": []
+            }  )"};
```

Comment:
I believe we can make this variable `const`:
```suggestion
    const auto test_parameters = Parameters{R"(
            {
                "model_part_name": "foo",
                "x_coordinates": [],
                "y_coordinates": []
            }  )"};
```

This suggestion also applies to the test parameters of the following test cases.

Reply:
Done